ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,321 @@
+*/
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using QuantConnect.Data.Custom.Benzinga;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Xml;
+
+namespace QuantConnect.ToolBox.Benzinga
+{
+    public class BenzingaDataConverter
+    {
+        private readonly DirectoryInfo _sourceDirectory;
+        private readonly DirectoryInfo _destinationDirectory;
+        private readonly bool _isWindows;
+        private readonly HashSet<string> _forbiddenTickers = new HashSet<string>()
+        {
+            "CON",
+            "PRN",
+            "AUX",
+            "NUL",
+            "COM1",
+            "COM2",
+            "COM3",
+            "COM4",
+            "COM5",
+            "COM6",
+            "COM7",
+            "COM8",
+            "COM9",
+            "LPT1",
+            "LPT2",
+            "LPT3",
+            "LPT4",
+            "LPT5",
+            "LPT6",
+            "LPT7",
+            "LPT8",
+            "LPT9"
+        };
+
+        /// <summary>
+        /// Creates a new instance of the Benzinga converter
+        /// </summary>
+        /// <param name="sourceDirectory">Directory to read data from. This should be the Benzinga data folder (e.g. /Data/alternative/benzinga)</param>
+        /// <param name="destinationDirectory">Directory to write data to. This should be the final Benzinga data folder (e.g. /Data/alternative/benzinga)</param>
+        public BenzingaDataConverter(DirectoryInfo sourceDirectory, DirectoryInfo destinationDirectory)
+        {
+            _sourceDirectory = sourceDirectory;
+            _destinationDirectory = destinationDirectory;
+            _destinationDirectory.Create();
+            _isWindows = OS.IsWindows;
+        }
+
+        public bool Convert(DateTime date)
+        {
+            var newsArticles = new DirectoryInfo(Path.Combine(
+                _sourceDirectory.FullName,
+                $"{date.Year}",
+                date.Month.ToStringInvariant().PadLeft(2, '0'),
+                date.Day.ToStringInvariant().PadLeft(2, '0')));
+
+            if (!newsArticles.Exists)
+            {
+                Log.Error($"BenzingaDataConverter.Convert(): Data not found for date {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            WriteToFile(Process(newsArticles), date);
+            return true;
+        }
+
+        /// <summary>
+        /// Process the data to BenzingaNews instances
+        /// </summary>
+        /// <param name="parentSourceDirectory">Source directory to read files from</param>
+        /// <param name="date"></param>
+        /// <returns></returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
```

Comment:
Should they be ordered in some way?

Reply:
The end result is the same regardless of ordering as we generate references to each article individually